habitat/datasets/rearrange/rearrange_generator.py

```diff
@@ -0,0 +1,1018 @@
+#!/usr/bin/env python3
+
+# Copyright (c) Facebook, Inc. and its affiliates.
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+import os
+import os.path as osp
+from typing import Any, Dict, List, Optional, Tuple
+
+import magnum as mn
+import numpy as np
+from tqdm import tqdm
+from yacs.config import CfgNode as CN
+
+import habitat.datasets.rearrange.samplers as samplers
+import habitat.datasets.rearrange.sim_utilities as sutils
+import habitat_sim
+from habitat.core.logging import logger
+from habitat.datasets.rearrange.rearrange_dataset import (
+    RearrangeDatasetV0,
+    RearrangeEpisode,
+)
+
+
+class RearrangeEpisodeGenerator:
```

Comment:
Should RearrangeEpisodeGenerator be inheriting a generic EpisodeGenerator? 
Can you add a comment on what an EpisodeGenerator does and how to use it?

Reply:
I'll add some better comments, but perhaps defining the API for an abstract EpisodeGenerator may be out of scope here. I generally agree we should follow-up with design and refactor. For example, `pointnav_generator.py` defines a classless API for generating episodes which should then be refactored to use a generator also.